crates/turbo-tasks-macros/src/value_trait_macro.rs

```diff
@@ -254,8 +262,13 @@ pub fn value_trait(args: TokenStream, input: TokenStream) -> TokenStream {
             }
         }
 
-        impl<T> #ident for T where #ref_ident: std::convert::From<T>, turbo_tasks::TaskInput: for<'a> std::convert::From<&'a T> #(, #supertrait_refs: std::convert::From<T>)* {
-            #(default #trait_fns)*
+        impl<T> #ident for T
+        where
+            T: turbo_tasks::ValueTraitVc,
+            #ref_ident: turbo_tasks::FromSubTrait<T>,
+            turbo_tasks::TaskInput: for<'a> std::convert::From<&'a T>
+            #(, #supertrait_refs: turbo_tasks::FromSubTrait<T>)* {
+            #(#trait_fns)*
         }
```

Comment:
This is the main change with the new `turbo_tasks::FromSubTrait` (just a copy paste of `std::convert::From`)

no more `default` impls